rules/TypeDeclaration/TypeInferer/SilentVoidResolver.php

```diff
@@ -41,11 +41,14 @@ public function hasExclusiveVoid(ClassMethod | Closure | Function_ $functionLike
             return false;
         }
 
-        $return = $this->betterNodeFinder->findFirstInFunctionLikeScoped(
-            $functionLike,
-            static fn (Node $node): bool => $node instanceof Return_ && $node->expr instanceof Expr
-        );
-        return ! $return instanceof Return_;
+        $returns = $this->betterNodeFinder->findInstancesOfInFunctionLikeScoped($functionLike, Return_::class);
+        foreach ($returns as $return) {
+            if ($return->expr instanceof Expr) {
+                return false;
+            }
+        }
+
+        return true;
```

Comment:
@staabm fixed with revert the logic from your PR:

- https://github.com/rectorphp/rector-src/pull/4923

as no returns at all need to be checked to return true.

Reply:
@staabm the `BetterNodeFinder::findFirstInFunctionLikeScoped()` seems needs refactor to ensure find next on scoped search, I created new PR for it to keep using `findFirstInFunctionLikeScoped()`

- https://github.com/rectorphp/rector-src/pull/4931